lib/requests.js

```diff
@@ -149,6 +158,12 @@ QueryRequest.prototype.write = function (encoder) {
   //v3: <query>
   //      <consistency><flags>[<n>[name_1]<value_1>...[name_n]<value_n>][<result_page_size>][<paging_state>][<serial_consistency>][<timestamp>]
   var frameWriter = new FrameWriter(types.opcodes.query);
+  var headerFlags = this.options.traceQuery ? types.frameFlags.tracing : 0;
```

Comment:
One thing I noticed that both the python and java drivers supports is that it uses Custom Payload returned from an associated Prepared Statement Response as the Custom Payload set on requests using that Prepared Statement if the User did not provide their own payload (see the algorithm defined [here](https://github.com/datastax/java-driver/blob/f6b5809d880dba65257298963c95da8387e9b9c7/features/custom_payloads/README.md#custom-payloads-and-prepared-statements)).

The way I think this could be supported here is whenever emitting a prepared statement you would also emit the customPayload with the id and metadata and pass it on to ExecuteRequest from Client._executeAsPrepared.

Do you think this is needed for the nodejs-driver?